include/seastar/core/gate.hh

```diff
@@ -50,8 +50,14 @@ class gate {
 public:
     gate() = default;
     gate(const gate&) = delete;
-    gate(gate&&) = default;
-    gate& operator=(gate&&) = default;
+    gate(gate&& x) noexcept
+        : _count(std::exchange(x._count, 0)), _stopped(std::exchange(x._stopped, std::nullopt)) {}
+    gate& operator=(gate&& x) noexcept {
+        assert(!_count && "gate reassigned with outstanding requests");
```

Comment:
If assigning a gate to itself, the resulting state would be broken

Reply:
yeah, added a check accordingly.